parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -689,55 +731,11 @@ loop = do
 --                      | r <- toList $ Names.typesNamed ns name ]
 --              in (terms, types)
 
-      LinkI src mdValue -> do
-        let srcle = toList (getHQ'Terms src)
-            srclt = toList (getHQ'Types src)
-            mdValuel = toList (getHQ'Terms mdValue)
-        case (srcle, srclt, mdValuel) of
-          (srcle, srclt, [Referent.Ref mdValue])
-            | length srcle < 2 && length srclt < 2 -> do
-              mdType <- eval $ LoadTypeOfTerm mdValue
-              case mdType of
-                Nothing -> respond $ LinkFailure input
-                Just ty -> do
-                  let parent = Path.toAbsolutePath currentPath' (fst src)
-                  let get = Branch.head <$> getAt parent
-                  before <- get
-                  stepAt (Path.unabsolute parent, step (Type.toReference ty))
-                  after <- get
-                  (ppe, outputDiff) <- diffHelper before after
-                  respondNumbered $ ShowDiffNamespace parent parent ppe outputDiff
-                where
-                step mdType b0 = let
-                  tmUpdates terms = foldl' go terms srcle where
-                    go terms src = Metadata.insert (src, mdType, mdValue) terms
-                  tyUpdates types = foldl' go types srclt where
-                    go types src = Metadata.insert (src, mdType, mdValue) types
-                  in over Branch.terms tmUpdates . over Branch.types tyUpdates $ b0
-          _ -> respond $ LinkFailure input
-
-      UnlinkI src mdValue -> do
-        let srcle = toList (getHQ'Terms src)
-            srclt = toList (getHQ'Types src)
-            (parent, _last) = resolveSplit' src
-            mdValuel = toList (getHQ'Terms mdValue)
-        case (srcle, srclt, mdValuel) of
-          (srcle, srclt, [Referent.Ref mdValue])
-            | length srcle < 2 && length srclt < 2 -> do
-              mdType <- eval $ LoadTypeOfTerm mdValue
-              case mdType of
-                Nothing -> respond $ LinkFailure input
-                Just ty -> do
-                  stepAt (parent, step (Type.toReference ty))
-                  success
```

Comment:
Was it intentional that the new definition produces some output that the old one doesn't?  Seems fine, just double-checking.